osgpkitools/incommon_request.py

```diff
@@ -60,172 +62,97 @@
+    optional.add_argument('-t', '--test', action='store_true', dest='test', default=False,
+                              help='Testing mode: test connection to InCommon API but does not request certificates. '
+                              'Useful to test authentication credentials, optional arguments are ignored.')
+    optional.add_argument('-v', '--version', action='version', version=utils.VERSION_NUMBER)
+
+    parsed_args = parser.parse_args()
+
+    # We can't add altnames to the mutually exclusive 'hosts' group since it's not a required opt
+    if parsed_args.hostfile and parsed_args.altnames:
+        parsed_args.altnames = []
+        print("-a/--altname option ignored with -F/--hostfile", file=sys.stderr)
+    
+    if parsed_args.debug:
         # this sets the root debug level
         logging.getLogger().setLevel(logging.DEBUG)
         logger.debug('Debug mode enabled')
     
-    if not args.login:
-        raise InsufficientArgumentException("InsufficientArgumentException: " + \
-                                            "Please provide the InCommon username (login)\n")
-        
-    if not args.test:    
-        if not args.hostname:
-            if args.hostfile is None:
-                raise InsufficientArgumentException("InsufficientArgumentException: " + \
-                                                    "Please provide hostname(-H) or file name containing hosts(-f)\n")
-            else:
-                    hostfile = args.hostfile
-        else:
-            hostname = args.hostname
-        
-    if not args.test:
-        if not args.hostname:
-            if not os.path.exists(hostfile):
-                raise FileNotFoundException(hostfile, 'Error: could not locate the hostfile')
-
-    arguments = dict()
-
-    if vars().has_key('args'):
-        arguments.update({'args': args})
-    if vars().has_key('values'):
-        arguments.update({'values': values})
-    if vars().has_key('hostname'):
-        arguments.update({'hostname': hostname})
-    
-    arguments.update({'alt_names': args.alt_names})
-    arguments.update({'test': args.test})
-    arguments.update({'login': args.login})
-    
-    if not args.usercert or not args.userprivkey:
+    # (-H/--hostname | -F/--hostfile) are mutually exclusive but not required so testing mode can be enabled with optional param -t/--test 
+    if not parsed_args.test and not parsed_args.hostname and not parsed_args.hostfile:
+        parser.print_usage()
         raise InsufficientArgumentException("InsufficientArgumentException: " + \
-                                            "Please provide certificate(-c, --cert) and key(-k, --pkey) files\n")
-    
-    usercert, userkey = utils.verify_user_cred(args.usercert, args.userprivkey)
-    arguments.update({'usercert': usercert})
-    arguments.update({'userprivkey': userkey})
+                                            "Please provide -H/--hostname and -F/--hostfile for normal mode. " + \
```

Comment:
This message should say "or", not "and".

Also I think it would be better to not mention the normal and testing modes or -t.  If somebody supplies -t you will just silently ignore -H and -F, that's fine.  If they don't supply -t, assume they want normal mode.

Reply:
Also the InsufficientArgumentException generates a different kind of message than other missing required options.  I recommend making it look the same, so the message is just the usage followed by "osg-incommon-cert-request: error: argument -H/--hostname or -F/--hostfile is required".

Reply:
Modified this line with parser.error.